model/metricset/event.go

```diff
@@ -206,21 +218,34 @@ func (me *Metricset) Transform(tctx *transform.Context) []beat.Event {
 		return nil
 	}
 
+	sampleSpanFields := common.MapStr{"span": common.MapStr{}}
+	sampleTransactionFields := common.MapStr{"transaction": common.MapStr{}}
+
 	fields := common.MapStr{}
 	for _, sample := range me.Samples {
-		if _, err := fields.Put(sample.Name, sample.Value); err != nil {
+		if utility.Contains(sample.Name, knownSpanSampleAttrs) {
+			sampleSpanFields.Put(sample.Name, sample.Value)
+		} else if utility.Contains(sample.Name, knownTransactionSampleAttrs) {
+			sampleTransactionFields.Put(sample.Name, sample.Value)
+		} else if _, err := fields.Put(sample.Name, sample.Value); err != nil {
 			logp.NewLogger("transform").Warnf("failed to transform sample %#v", sample)
 			continue
 		}
 	}
 
+	transactionFields := me.Transaction.fields()
+	transactionFields.DeepUpdate(sampleTransactionFields["transaction"].(common.MapStr))
+
+	spanFields := me.Span.fields()
+	spanFields.DeepUpdate(sampleSpanFields["span"].(common.MapStr))
+
 	fields["processor"] = processorEntry
 	tctx.Metadata.Set(fields)
 
 	// merges with metadata labels, overrides conflicting keys
 	utility.DeepUpdate(fields, "labels", me.Labels)
-	utility.DeepUpdate(fields, transactionKey, me.Transaction.fields())
-	utility.DeepUpdate(fields, spanKey, me.Span.fields())
+	utility.DeepUpdate(fields, transactionKey, transactionFields)
+	utility.DeepUpdate(fields, spanKey, spanFields)
```

Comment:
Why are the changes in the transformation method necessary? I see you are changing the order of merging the data, but I don't see any specific reason for one or the other. Apart from that I don't see why those changes would be necessary and the code gets more complicated than before.

Reply:
Not sure I understand your question. 
`transaction` and `span` fields inside `samples` need to be pulled up one level, so I collect them in L227 and L229 and move them in L237 and L240, for the result that you can see in the approved files.
Other ideas of how to achieve the same?

> you are changing the order of merging the data

Where am I doing that?

Reply:
Current master already pulls out span and transaction information from samples and merges it with top level span and transaction information. That has been done in PR #2265 and #2287 . Therefore I don't understand why this method needs to be changed at all.
From my understanding what was missing is the validation of defined metric attributes on the Intake API level and the mappings in the templates. 
Please correct me if I'm wrong, I might overlook something, but that's my understanding from this. 


>>you are changing the order of merging the data

>Where am I doing that?

so far 
L211: `if _, err := fields.Put(sample.Name, sample.Value); err != nil ` has put span and transaction info from sample to root span and transaction keys
And then information that had been sent top level has been deep merged to them:
L222: `utility.DeepUpdate(fields, transactionKey, me.Transaction.fields())`
L223: `utility.DeepUpdate(fields, spanKey, me.Span.fields())`

Now you first take the root level span and transaction fields and then deep update with the information extracted from samples. 

I don't think the order matters, but as it shouldn't matter here I would go with the existing simpler solution.

Reply:
> Current master already pulls out span and transaction information from samples

I missed that, then. Ill have another look, thanks